music21/chord/__init__.py

```diff
@@ -47,9 +47,339 @@ class ChordException(exceptions21.Music21Exception):
+        >>> c1 == n1
+        False
+        >>> c2.duration.quarterLength = 2.0
+        >>> c1 == c2
+        False
+        >>> c1 != c2
+        True
+        '''
+        if not super().__eq__(other):
+            return False
+        if not hasattr(other, 'notes'):
+            return False
+        if not len(self.notes) == len(other.notes):
+            return False
+        return True
+
+    def __deepcopy__(self: _ChordType, memo=None) -> _ChordType:
+        '''As Chord objects have one or more Volume, objects, and Volume
+        objects store weak refs to the to client object, need to specialize
+        deepcopy handling depending on if the chord has its own volume object.
+        '''
+        # environLocal.printDebug(['calling NotRest.__deepcopy__', self])
+        # as this inherits from NotRest, can use that __deepcopy__ as basis
+        # that looks only to _volume to see if it is not None; with a
+        # Chord, _volume will always be None
+        new = super().__deepcopy__(memo=memo)
+        # after copying, if a Volume exists, it is linked to the old object
+        # look at _volume so as not to create object if not already there
+        # noinspection PyProtectedMember
+        for d in new._notes:
+            # if .volume is called, a new Volume obj will be created
+            if d.hasVolumeInformation():
+                d.volume.client = new  # update with new instance
+        return new
+
+    # TODO: __getitem__
+
+    def __iter__(self):
+        return iter(self._notes)
+
+    def __len__(self):
+        '''
+        Return the length of components in the chord.
+
+        >>> c = chord.Chord(['c', 'e', 'g'])
+        >>> len(c)
+        3
+        '''
+        return len(self._notes)
+
+    def _add_core_or_init(self, notes, *, useDuration=None):
+        '''
+        This is the private append method called by .add and called by __init__.
+
+        It differs from the public method in that a duration object can
+        be passed in which is used for the first note of the chord or as many pitches
+        as can use it -- it's all an optimization step to create as few duration objects
+        as is necessary.
+
+        Also requires that notes be an iterable.
+        '''
+        # quickDuration specifies whether the duration object for the chord
+        # should be taken from the first note of the list.
+        quickDuration = False
+        if useDuration is None:
+            useDuration = self.duration
+            quickDuration = True
+
+        for n in notes:
+            if isinstance(n, pitch.Pitch):
+                # assign pitch to a new Note
+                if useDuration:  # not False or None
+                    newNote = note.Note(n, duration=useDuration)
+                else:
+                    newNote = note.Note(n)
+                self._notes.append(newNote)
+                # self._notes.append({'pitch':n})
+            elif isinstance(n, Chord):
```

Comment:
probably BaseChord?  So that one Unpitched can merge another?

Reply:
Great catch!